File: head-first/chapter01/SimUDuckKotlin/src/main/kotlin/duck/MallardDuck.kt
package duck

import fly.FlyWithWings
import quack.Quack

class MallardDuck() : Duck() {

    init {
        quackBehavior = Quack()
        flyBehavior = FlyWithWings()
    }

    override fun display() {
        println("물오리")
    }
}

File: head-first/chapter01/simuduck-dart/lib/duck/MallardDuck.dart

import 'package:simuduck_dart/fly/FlyWithWings.dart';
import 'package:simuduck_dart/quack/Quack.dart';

import 'Duck.dart';

class MallardDuck extends Duck {

  MallardDuck() {
    flyBehavior = FlyWithWings();
    quackBehavior = Quack();
  }

  @override
  void display() {
    print('물오리');
  }
}